### DISCUSSION TITLE
Seeding Data

### DISCUSSION THREAD

**Discussion by whymlaszlo** [supabase/supabase#19773] [Feature Requests] [answered]
Hello all,

The current situation is that you have to write a seed script if you want to have seeding data in the database. This is then located here:  `./supabase/seed.sql`
However, the script can quickly become quite large and confusing. 
Is it perhaps possible to define a location, e.g. `./supabase/seed/`, where all `.sql` files are then read in and processed in sequence? This would make it possible to write a script `01_users.sql`, which first writes the users to the database and then reads a `02_orders.sql `and writes orders to the database. This may be much more structured.
What do you think? Or is there already a similar functionality?

Thanks.

**Comment by jp30566347** [comment]
Totally, I really want this feature! 

I'm willing to contribute if someone could point me in the right direction. :P

**Comment by ranielzed** [comment]
Tying this request with branching feature. It would be nice to have an option to configure seed directories for each of the branches . 
Say, `staging-branch` would have seeds defined in `./supabase/seed/staging-branch` or something similar.

**Comment by avallete** [comment, marked as answer]
Hi everyone,

I'm excited to share that we have some news on this.
You should now be able to do this via the `[remotes.branch-name.db.seed]` feature. 

For example, if you have a staging branch, you can set up your configuration like this:

```toml
[remotes.staging]
project_id = "your-staging-project-ref"

[remotes.staging.db.seed]
sql_paths = [
  "./seeds/staging/01_users.sql",
  "./seeds/staging/02_orders.sql"
]
```

This configuration tells Supabase branching to configure some seed specifics for the staging branch. Or any other branch.

For more details, please check out the [remote config docs](https://supabase.com/docs/guides/local-development/cli/config#remotes.branch_name.project_id) and the [db seeding docs](https://supabase.com/docs/guides/local-development/seeding-your-database#splitting-up-your-seed-file)

Let us know if you have any feedback or questions. Happy coding!
